docs/spec2006-tutorial.md

```diff
@@ -0,0 +1,517 @@
+    cwd = 'disk-image/',
+    path = 'disk-image/spec2006/spec2006-image/spec2006',
+    inputs = [packer, experiments_repo, m5_binary,],
+    documentation = 'Ubuntu Server with SPEC 2006 installed, m5 binary installed and root auto login'
+)
+```
+### Compiling Linux Kernel  
+In this step, we will download Linux kernel source code and compile the Linux kernel. 
+The file of interest in this step is the vmlinux file.  
+
+First, we download the Linux kernel source code. 
+The following command will shallow clone the linux stable repository as well as checking out the tag v4.19.83, which contains the code for linux kernel version 4.19.83. 
+This also works well for other version numbers.  
+
+In the root of the experiment folder,  
+```sh
+git clone --branch v4.19.83 --depth 1 https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/
+mv linux linux-4.19.83
+```
+Now, in launch_spec_experiments.py, we make an Artifact object of the Linux stable git repo.
+```python
+linux_repo = Artifact.registerArtifact(
+    command = '''git clone git clone --branch v4.19.83 --depth 1 https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/;
+    mv linux linux-4.19.83''',
+    typ = 'git repo',
+    name = 'linux-4.19.83',
+    path =  'linux-4.19.83',
+    cwd = './',
+    documentation = 'Linux kernel 4.19 source code repo obtained in November'
+)
+```
+Next, we compile the Linux kernel. 
+We will make a folder named linux-configs containing all working linux configs. 
+Working Linux configs and documentations for generating a Linux config for a Linux version could be found [here](https://github.com/darchr/gem5art/tree/master/docs/linux-configs).  
+
+In the root folder of the experiment,
+```sh
+mkdir linux-configs
+```
+To download the linux-4.19.83 configs,
+```sh
+cd linux-configs
+wget https://raw.githubusercontent.com/darchr/gem5art/master/docs/linux-configs/config.4.19.83
+```
+The following commands will copy the linux config and compile the linux kernel. 
+In the root folder of the experiment,
+```sh
+cp linux-configs/config.4.19.83 linux-4.19.83/.config
+cd linux-4.19.83
+make -j8
+cp vmlinux vmlinux-4.19.83
+```
+Now, in launch_spec_experiments.py, we make an Artifact object of the Linux kernel binary.
+```python
+linux_binary = Artifact.registerArtifact(
+    name = 'vmlinux-4.19.83',
+    typ = 'kernel',
+    path = 'linux/vmlinux-4.19.83',
+    cwd = './',
+    command = '''cp linux-configs/config.4.19.83 linux-4.19.83/.config
+	cd linux-4.19.83
+	make -j8
+	cp vmlinux vmlinux-4.19.83
+    ''',
+    inputs = [experiments_repo, linux_repo,],
+    documentation = "kernel binary for v4.19.83",
+)
+```
+### The gem5 Run Script/gem5 Configuration
+In this step, we take a look at the final missing piece: the gem5 run script. 
+The script is where we specify the simulated system. 
+We offer example scripts in the [configs-spec2006-tests folder](https://github.com/darchr/gem5art/blob/master/docs/configs-spec2006-tests/).   
```

Comment:
I'm not sure this name (`configs-spec2006-tests`) is the same as what was discussed above in the tutorial.